R/constructors.R

```diff
@@ -141,6 +142,12 @@ new_quant_param <- function(type = c("double", "integer"),
     }
   }
 
+  if (lifecycle::is_present(default)) {
```

Comment:
I think this check should actually be the first thing you do in the function. I like to have deprecation behavior be wayyyy at the top of a function - it makes it clear to the reader that that argument isn't going to be important